crates/secrets-types/src/lib.rs

```diff
@@ -196,19 +208,19 @@ impl TryInto<HashMap<String, String>> for SecretConfig {
         if let Some(version) = self.version {
             map.insert("version".to_string(), version);
         }
-        let policy_properties = HashMap::from([
-            ("type".to_string(), SECRET_POLICY_TYPE.to_string()),
+
+        let policy = HashMap::from([
+            (
+                "type".to_string(),
+                serde_json::Value::String(SECRET_POLICY_PROPERTIES_TYPE.to_string()),
+            ),
             (
                 "properties".to_string(),
-                serde_json::to_string(&self.policy_properties)
-                    .context("failed to serialize policy_properties map as string")?,
+                serde_json::to_value(&self.policy_properties)
+                    .context("failed to serialize policy properties")?,
             ),
         ]);
-        map.insert(
-            "policy_properties".to_string(),
-            serde_json::to_string(&policy_properties)
-                .context("failed to serialize policy_properties map as string")?,
-        );
+        map.insert("policy".to_string(), serde_json::to_string(&policy)?);
```

Comment:
I do kind of wonder why this isn't in the `::from()` above -- and also it might be nice to add a `.context`

Reply:
Just because they are separate objects, but I can add that `.context`